src/main/java/org/jvnet/hudson/plugins/repositoryconnector/ArtifactResolver.java

```diff
@@ -90,12 +99,24 @@ public boolean enableRepoLogging() {
     public List<Artifact> getArtifacts() {
         return artifacts;
     }
+    
+    public Collection<Repository> getRepos(AbstractBuild<?,?> build, String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if(repoid.compareTo("ALL")==0 || repoid.compareTo(e.getKey())==0) {
```

Comment:
thats a strange way to check strings for equals... why not like this:
`if("ALL".equals(repoid) || repoid.equals(e.getKey()))` ?